test/extension/handlers/topologymutation/handler.go

```diff
@@ -19,70 +19,239 @@ package topologymutation
 	log := ctrl.LoggerFrom(ctx)
-	log.Info("GeneratePatches called")
 
+	log.Info("GeneratePatches called")
 	walkTemplates(h.decoder, req, resp, func(obj runtime.Object, variables map[string]apiextensionsv1.JSON) error {
-		if dockerClusterTemplate, ok := obj.(*infrav1.DockerClusterTemplate); ok {
-			if err := patchDockerClusterTemplate(dockerClusterTemplate, variables); err != nil {
-				return err
+		log.Info("Patching template", "template", obj.GetObjectKind().GroupVersionKind().String())
+
+		switch typedObj := obj.(type) {
+		case *infrav1.DockerClusterTemplate:
+			if err := patchDockerClusterTemplate(ctx, typedObj, variables); err != nil {
+				log.Error(err, "error patching DockerClusterTemplate")
+				return errors.Wrap(err, "error patching DockerClusterTemplate")
+			}
+		case *controlplanev1.KubeadmControlPlaneTemplate:
+			if err := patchKubeadmControlPlaneTemplate(ctx, typedObj, variables); err != nil {
+				log.Error(err, "error patching KubeadmControlPlaneTemplate")
+				return errors.Wrapf(err, "error patching KubeadmControlPlaneTemplate")
 			}
-		}
 
+		case *bootstrapv1.KubeadmConfigTemplate:
+			if err := patchKubeadmConfigTemplate(ctx, typedObj, variables); err != nil {
+				log.Error(err, "error patching KubeadmConfigTemplate")
+				return errors.Wrap(err, "error patching KubeadmConfigTemplate")
+			}
+		case *infrav1.DockerMachineTemplate:
+			if err := patchDockerMachineTemplate(ctx, typedObj, variables); err != nil {
+				log.Error(err, "error patching DockerMachineTemplate")
+				return errors.Wrap(err, "error patching DockerMachineTemplate")
+			}
+		default:
+			return nil
+		}
 		return nil
 	})
 }
 
-// patchDockerClusterTemplate patches the DockerClusterTepmlate.
-func patchDockerClusterTemplate(dockerClusterTemplate *infrav1.DockerClusterTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
-	// Get the variable value as JSON string.
-	value, err := patchvariables.GetVariableValue(templateVariables, "lbImageRepository")
+func patchKubeadmControlPlaneTemplate(ctx context.Context, kubeadmControlPlaneTemplate *controlplanev1.KubeadmControlPlaneTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	// 1) If the Kubernetes version from builtin.controlPlane.version is below 1.24.0 set "cgroup-driver": "cgroupfs" to
+	//    - kubeadmConfigSpec.InitConfiguration.KubeletExtraArgs
+	//    - kubeadmConfigSpec.JoinConfiguration.KubeletExtraArgs
+	_, cpVersion, err := getStringValueForVariable(templateVariables, "builtin.controlPlane.version")
+	if err != nil {
+		return errors.Wrap(err, "could not set cgroup-driver to control plane template kubeletExtraArgs")
+	}
+	controlPlaneVersion, err := version.ParseMajorMinorPatchTolerant(cpVersion)
 	if err != nil {
 		return err
 	}
+	if version.Compare(controlPlaneVersion, versionCeilingForPatch) == -1 {
+		log.Info(fmt.Sprintf("Setting kubeadmControlPlaneTemplate cgroup-driver %s", "cgroup-driver"))
```

Comment:
```suggestion
		log.Info(fmt.Sprintf("Setting kubeadmControlPlaneTemplate cgroup-driver to %q", cgroupDriverCgroupfs))
```
or something, but let's use the constant (cgroup-driver vs cgroupfs)

P.S. I think it reads better with "to" (would also apply to other logs below)